index.js

```diff
@@ -157,6 +152,92 @@ app.get('/logout', (req, res) => {
     }
 });
 
+app.get('/dashboard', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+
+    try {
+        let allLocations = await db.getLocations()
+        
+        let viewParams = {
+            locations: allLocations
+                .map(location => { 
+                    return { name: location.displayName, id: location.locationid }
+                })
+        }
+        viewParams.viewMessage = allLocations.length >= 1 ? 'Please select a location' : 'No locations to display'
+
+        res.send(Mustache.render(locationsDashboardTemplate, viewParams))
+    }
+    catch(err) {
+        console.log(err)
+        res.status(500).send()
+    }
+})
+
+app.get('/dashboard/:locationId', async (req, res) => {
+    if (!req.session.user || !req.cookies.user_sid) {
+        res.redirect('/login')
+        return
+    }
+    else if(typeof req.params.locationId !== "string") {
+        let locations = await db.getlocations()
```

Comment:
I think the "l" in `getLocations` should be capitalized.

Reply:
I'm actually not 100% sure this needs to be here, it was copied in from buttons but I think this has been circumvented by my including a landing page at `/dashboard`?

Reply:
Hmmm, yeah, I don't know when we'd end up in a case where `locationId` from the URL would ever not be a string.

Reply:
Yeah, I think it was to avoid having a landing page and just go to the first thing in the list, which is what was breaking it when there were no installations in buttons 🤣 I've taken it out for now